forge-gui/res/cardsfolder/upcoming/forensic_gadgeteer.txt

```diff
@@ -0,0 +1,8 @@
+Name:Forensic Gadgeteer
+ManaCost:2 U
+PT:2/3
+Types:Creature Vedalken Artificer Detective
+T:Mode$ SpellCast | ValidCard$ Artifact | ValidActivatingPlayer$ You | Execute$ TrigInvestigate | TriggerZones$ Battlefield | TriggerDescription$ Whenever you cast an artifact spell, investigate
+SVar:TrigInvestigate:DB$ Investigate
+S:Mode$ ReduceCost | ValidCard$ Artifact | ValidActivatingPlayer$ You | Type$ Ability | Amount$ 1 | MinMana$ 1 | AffectedZone$ Battlefield | Description$ Activated abilities of artifacts you control {1} less to activate. This effect can't reduce the mana in that cost to less than one mana.
```

Comment:
it needs to be scripted more like on _Training Grounds_
it matters when e. g. _Armageddon Clock_ would be in play
see the difference?

Reply:
Removed `ValidActivatingPlayer$ You` constraint and replaced `Artifact` with `Artifact.YouCtrl` for the `ValidCard$` constraint.

Tested this against a deck with Gadgeteer, Armageddon Clock and Well of Knowledge. Can verify I can activate their clock and wells at reduced costs if they have a Gadgeteer in play.